packages/react-lowcode/src/codegen/generation/generators/template/template-resolver.ts

```diff
@@ -0,0 +1,77 @@
+import ts, { factory } from "typescript";
+import { printSourceCode } from "../../../ast/ast";
+import { createImportDeclaration } from "../../../ast/imports";
+import { createAst } from "../../code-generation/createSourceFile";
+import { Entity } from "../../entity";
+import { EntityHelper } from "../../entity/helper";
+
+export default class TemplateResolver {
+    private _entity?: Entity
+
+    constructor(entity?: Entity) {
+        this._entity = entity;
+    }
+
+    generateListPage(template: string): string | undefined {
+        let result: string | undefined;
+
+        if (this._entity) {
+            let ast = createAst(template);
+            
+            if (ast) {
+                const tableComponentName = EntityHelper.getTableComponentName(this._entity);
+                const inputParameterIdentifier = EntityHelper.getInputParameterIdentifier(this._entity);
+
+                const tableComponentElement = factory.createJsxSelfClosingElement(
+                    factory.createIdentifier(tableComponentName),
+                    undefined,
+                    factory.createJsxAttributes([factory.createJsxAttribute(
+                        inputParameterIdentifier,
+                        factory.createJsxExpression(
+                            undefined,
+                            factory.createPropertyAccessChain(
+                                factory.createIdentifier("data"),
+                                factory.createToken(ts.SyntaxKind.QuestionDotToken),
+                                inputParameterIdentifier
+                            )
+                        )
+                    )])
+                );
+
+                const transformer = <T extends ts.Node>(): ts.TransformerFactory<T> => {
+                    return context => {
+                        const visit: ts.Visitor = node => {
+                            if (ts.isImportDeclaration(node)) {

```

Comment:
the code will be more readable (intutive) if you extract this code to function
`replaceImportDeclaration('ListPlaceholder', tableComponenetName, tableComponentFrom)`
`replaceTag('ListPlaceholder', tableComponentElement)`

Reply:
Yes, it would be intuitive. 
I focused on modifying both the import and the element with one cycle of searching all the elements in the template. Using 2 functions I have to search all elements in the template 2x.

However, I can change that, in this case, it probably won't have any effect on performance. 
What do you think?

Reply:
I adjusted it a bit (all changes are made within one transformation (with one cycle of searching all the elements in the template)).